src/commands/project/convert/source-behavior.ts

```diff
@@ -0,0 +1,105 @@
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { Messages } from '@salesforce/core';
+import {
+  getValidatedProjectJson,
+  TMP_DIR,
+  convertToMdapi,
+  DRY_RUN_DIR,
+  PRESETS_PROP,
+  PRESET_CHOICES,
+  getPackageDirectoriesForPreset,
+  convertBackToSource,
+} from '../../../utils/convertBehavior.js';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-deploy-retrieve', 'convert.source-behavior');
+
+export type SourceBehaviorResult = {
+  [PRESETS_PROP]: string[];
+  deletedFiles: string[];
+  createdFiles: string[];
+};
+
+export default class ConvertSourceBehavior extends SfCommand<SourceBehaviorResult> {
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+  public static readonly state = 'beta';
+  public static readonly requiresProject = true;
+
+  public static readonly flags = {
+    behavior: Flags.option({
+      summary: messages.getMessage('flags.behavior.summary'),
+      char: 'b',
+      required: true,
+      options: PRESET_CHOICES,
+    })(),
+    'dry-run': Flags.boolean({
+      summary: messages.getMessage('flags.dry-run.summary'),
+    }),
+    'preserve-temp-dir': Flags.boolean({
+      summary: messages.getMessage('flags.preserve-temp-dir.summary'),
+    }),
+    'target-org': Flags.optionalOrg(),
+  };
+
+  public async run(): Promise<SourceBehaviorResult> {
+    const { flags } = await this.parse(ConvertSourceBehavior);
+    if (await flags['target-org']?.supportsSourceTracking()) {
+      throw messages.createError('error.trackingNotSupported');
+    }
+    const projectJson = getValidatedProjectJson(flags.behavior, this.project!);
+    const [backupPjsonContents, packageDirsWithDecomposable] = await Promise.all([
+      flags['dry-run'] ? readFile(projectJson.getPath()) : '',
+      getPackageDirectoriesForPreset(this.project!, flags.behavior),
+    ]);
+    const filesToDelete = await convertToMdapi(packageDirsWithDecomposable);
+
+    // flip the preset in the sfdx-project.json, even for dry-run, since the registry will need for conversions
+    projectJson.set(PRESETS_PROP, [...(projectJson.get<string[]>(PRESETS_PROP) ?? []), flags.behavior]);
+    await projectJson.write();
+    this.info(`sfdx-project.json ${PRESETS_PROP} is now [${projectJson.get<string[]>(PRESETS_PROP).join(',')}]`);
+
+    // delete the “original” files that no longer work because of project update
+    await Promise.all(flags['dry-run'] ? [] : filesToDelete.map((f) => rm(f)));
+
+    const createdFiles = await convertBackToSource({
+      packageDirsWithPreset: packageDirsWithDecomposable,
+      projectDir: this.project!.getPath(),
+      dryRun: flags['dry-run'],
+    });
+
+    if (!flags['preserve-temp-dir']) {
+      await rm(TMP_DIR, { recursive: true });
```

Comment:
🔴 This is not working, when I use `--dry-run` without `--preserve-temp-dir` I have a directory call `DRY-RUN-RESULTS`. It looks like the value of `TMP_DIR` is `decompositionConverterTempDir`

Reply:
yes, intentional.  `DRY-RUN-RESULTS` is not the TMP_DIR.

You'd want to see what would have ended up in your project if you use `--dry-run`.  Not just the filenames, but the contents of those files.

Reply:
I see. I do see the distinction now. I added a simple change to the `--help` to specify where the dry-run files will be created
https://github.com/salesforcecli/plugin-deploy-retrieve/pull/1015/files#r1617872864